src/extractor/guidance/intersection_generator.cpp

```diff
@@ -123,31 +135,71 @@ IntersectionGenerator::ComputeIntersectionShape(const NodeID node_at_center_of_i
             BOOST_ASSERT(std::abs(bearing) <= 0.1);
         }
 
-        intersection.push_back({edge_connected_to_intersection, bearing, segment_length});
+        initial_roads_ordering.push_back(
+            {initial_bearing, {edge_connected_to_intersection, bearing, segment_length}});
     }
 
-    if (!intersection.empty())
+    if (!initial_roads_ordering.empty())
     {
-        const auto base_bearing = [&]() {
+        const auto base_initial_bearing = [&]() {
             if (sorting_base)
             {
-                const auto itr =
-                    std::find_if(intersection.begin(),
-                                 intersection.end(),
-                                 [&](const IntersectionShapeData &data) {
-                                     return node_based_graph.GetTarget(data.eid) == *sorting_base;
-                                 });
-                if (itr != intersection.end())
+                const auto itr = std::find_if(initial_roads_ordering.begin(),
+                                              initial_roads_ordering.end(),
+                                              [&](const auto &data) {
+                                                  return node_based_graph.GetTarget(
+                                                             data.road.eid) == *sorting_base;
+                                              });
+                if (itr != initial_roads_ordering.end())
                     return util::bearing::reverse(itr->bearing);
             }
-            return util::bearing::reverse(intersection.begin()->bearing);
+            return util::bearing::reverse(initial_roads_ordering.begin()->bearing);
         }();
-        std::sort(intersection.begin(),
-                  intersection.end(),
-                  makeCompareShapeDataAngleToBearing(base_bearing));
+
+        // sort roads with respect to the initial bearings, tie-breaker for equal initial bearings
+        // is to order CCW road before CW one
```

Comment:
@oxidase personally I do not understand what wold be a cw road/ccw road. Could you clarify here a bit?